File: src/2025-02-23-deepclaude-try/cpp/exercises/practice/grade-school/grade_school.cpp
#include "grade_school.h"
#include <algorithm>

namespace grade_school {

void school::add(const std::string& name, int grade) {
    // Check if student already exists
    auto it = std::find_if(students_.begin(), students_.end(),
        [&name](const Student& s) { return s.name == name; });
    
    if (it == students_.end()) {
        students_.push_back({name, grade});
        std::sort(students_.begin(), students_.end());
    }
}

std::vector<std::string> school::roster() const {
    std::vector<std::string> result;
    result.reserve(students_.size());
    
    for (const auto& student : students_) {
        result.push_back(student.name);
    }
    
    return result;
}

std::vector<std::string> school::grade(int grade_number) const {
    std::vector<std::string> result;
    
    for (const auto& student : students_) {
        if (student.grade == grade_number) {
            result.push_back(student.name);
        }
    }
    
    std::sort(result.begin(), result.end());
    return result;
}

}  // namespace grade_school

File: src/2025-02-23-deepclaude-try/javascript/exercises/practice/grade-school/grade-school.js
//
// This is only a SKELETON file for the 'Grade School' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class GradeSchool {
  constructor() {
    this._roster = {};
    this._studentGrades = new Map();
  }

  roster() {
    const sortedGrades = Object.keys(this._roster)
      .map(Number)
      .sort((a, b) => a - b);
    
    const result = {};
    for (const grade of sortedGrades) {
      result[grade] = [...this._roster[grade]];
    }
    
    return result;
  }

  add(name, grade) {
    if (this._studentGrades.has(name)) {
      const previousGrade = this._studentGrades.get(name);
      const index = this._roster[previousGrade].indexOf(name);
      if (index > -1) {
        this._roster[previousGrade].splice(index, 1);
      }
      this._studentGrades.delete(name);
    }

    if (!this._roster[grade]) {
      this._roster[grade] = [];
    }
    
    this._roster[grade].push(name);
    this._roster[grade].sort((a, b) => a.localeCompare(b));
    this._studentGrades.set(name, grade);
  }

  grade(gradeNumber) {
    const students = this._roster[gradeNumber] || [];
    return [...students];
  }
}
